### PR TITLE
Added protocols/tableview post.

### PR DESCRIPTION
**Pull request by ashfurrow**
I had no internet on the ferry last night, so I wrote this (the tweets were still cached in TweetBot). Added some GIFs and BLAMO here we are. 

Things I'm uncertain about:

- Using the `: class` suffix on protocol declarations. Good? Bad? First time I've used it. 
- GIFS – too many? Not enough? 
- I also tried using `inout` parameters but it felt suuuuper weird. Anyone have feels?

### CODE REVIEW

File: source/blog/2015-01-30-protocols-and-swift.html.markdown [hunk 1]
```diff
@@ -0,0 +1,328 @@
+---
+title: Protocols and Swift
+date: 2015-01-30 12:36:01 UTC
+---
+
+The other day I posed this (somewhat cryptic) question to Twitter: 
+
+<!-- more -->
+
+<blockquote class="twitter-tweet" lang="en"><p>Best way for pure Swift objects to have optional conformance to some contract? Optional protocols/protocol members are <a href="https://twitter.com/objc">@objc</a>-only.</p>&mdash; Ash Furrow (@ashfurrow) <a href="https://twitter.com/ashfurrow/status/560436119956115456">January 28, 2015</a></blockquote> <script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
```
Comment (a2) [inline source/blog/2015-01-30-protocols-and-swift.html.markdown]: Doesn't seem necessary to link to the @​objc Twitter user.

**Comment by a2** [comment]
pusheen_party

**Comment by alexito4** [comment]
`Yes, but protocols are like XML` 😂

Reading the conclusion I was thinking how cool it would be to do:

    extension FoodConsumer {
        func eat(food: Food, grams: Double) -> FoodConsumer {
           return food.beConsumedBy(self, grams: grams)
        }
    }

Iep, having a default implementation for every FoodConsummer (the protocol). I think that the Dog struct will also implement the eat almost in the same way, so this would be useful.

**Comment by ashfurrow** [comment]
@alexito4 I agree, but unfortunately protocols can't be extended. This is probably why we have 

    func countElements<T : _CollectionType>(x: T) -> T.Index.Distance

Instead of 

```swift
extension _CollectionType {
    var count: Int { get }
}
```

<div style="text-align: center; font-size: 110px">:/</div>

**Comment by ashfurrow** [comment]
Maybe including a comment about that would be helpful, though.

**Comment by saniul** [comment]
Chris Lattner mentioned that it’s a deficiency, so it’s very likely that a future version of Swift will allow us to rely on functions less. [Source](https://github.com/ksm/SwiftInFlux#moving-functionality-from-global-functions-to-methods)

**Comment by saniul** [comment]
Also https://github.com/ashfurrow/blog/pull/43

**Comment by ashfurrow** [comment]
Thanks @saniul – that'll make a great addition to the post. And thank you for the Pull Request.

**Comment by jspahrsummers** [comment]
Nice, I really like this post! :sparkles:

**Pull request merged**